var/spack/repos/builtin/packages/py-fs/package.py

```diff
@@ -13,6 +16,18 @@ class PyFs(PythonPackage):
     pypi     = "fs/fs-2.4.14.tar.gz"
 
     version('2.4.14', sha256='9555dc2bc58c58cac03478ac7e9f622d29fe2d20a4384c24c90ab50de2c7b36c')
+    version('0.5.4', sha256='ba2cca8773435a7c86059d57cb4b8ea30fda40f8610941f7822d1ce3ffd36197')
+
+    def patch(self):
+        if self.spec.satisfies('@0.5.5:'):
+            return
+
+        # Newer setuptools will complain that '2to3' is not a valid phase
+        filter_file('PY3 =.*', 'PY3 = False', 'setup.py')
```

Comment:
Should we just limit the version of setuptools to older versions?

Reply:
Until build deps are concretized separately, this will break environments where newer setuptools are required

Reply:
Looking at our current packages, there seem to be almost no packages that require setuptools 58+ and many packages that require older setuptools, so I think this is a non-issue at the moment:

<details>
 <summary>Click to expand</summary>

```console
$ grep depends_on..py-setuptools@ */package.py
ambari/package.py:    depends_on('py-setuptools@:44', type='build')
aomp/package.py:    depends_on('py-setuptools@:44', when='@3.5.0:3.8.0',
berkeleygw/package.py:    depends_on('py-setuptools@:44', type=('build', 'run'), when='+python')
gaudi/package.py:    depends_on('py-setuptools@:45', when='^python@:2.7', type='build')
gdal/package.py:    depends_on('py-setuptools@:57', type='build', when='@:3.0+python')
geopm/package.py:    depends_on('py-setuptools@39.2.0:', when="@1.0.0:", type='build')
gpdb/package.py:    depends_on('py-setuptools@:44')
open3d/package.py:    depends_on('py-setuptools@40.8:', when='+python', type='build')
paraview/package.py:    depends_on('py-setuptools@:44', when='+python')
py-aiohttp-cors/package.py:    depends_on('py-setuptools@20.8.1:', type='build')
py-anuga/package.py:    depends_on('py-setuptools@:44',      type=('build'),        when='@:2.1')
py-anyio/package.py:    depends_on('py-setuptools@42:', type='build')
py-apipkg/package.py:    depends_on('py-setuptools@30.3.0:', type='build')
py-apscheduler/package.py:    depends_on('py-setuptools@0.7:', type='build')
py-astroid/package.py:    depends_on('py-setuptools@17.1:', type='build')
py-astroid/package.py:    depends_on('py-setuptools@20:', type='build', when='@2.7.3:')
py-astropy-helpers/package.py:    depends_on('py-setuptools@30.3:', type='build')
py-backports-shutil-get-terminal-size/package.py:    depends_on('py-setuptools@:30,41:', type='build')
py-bcolz/package.py:    depends_on('py-setuptools@18.1:', type='build')
py-bcrypt/package.py:    depends_on('py-setuptools@40.8:', when='@3.1.7:', type='build')
py-boost-histogram/package.py:    depends_on('py-setuptools@45:', type='build')
py-brian2/package.py:    depends_on('py-setuptools@21:', type=('build', 'run'))
py-brian2/package.py:    depends_on('py-setuptools@24.2:', type=('build', 'run'), when='@2.4:')
py-cachetools/package.py:    depends_on('py-setuptools@46.4.0:', type='build', when='@4.2.2:')
py-cairocffi/package.py:    depends_on('py-setuptools@39.2.0:', type='build')
py-cartopy/package.py:    depends_on('py-setuptools@0.7.2:', type='build')
py-cftime/package.py:    depends_on('py-setuptools@18.0:', type='build')
py-cmake/package.py:    depends_on('py-setuptools@42:', type='build')
py-correctionlib/package.py:    depends_on('py-setuptools@42:', type='build')
py-crcmod/package.py:    depends_on('py-setuptools@40.0.0:', type='build')
py-cryptography/package.py:    depends_on('py-setuptools@40.6:', when='@2.7:', type='build')
py-cryptography/package.py:    depends_on('py-setuptools@18.5:', when='@2.2:2.6', type='build')
py-cryptography/package.py:    depends_on('py-setuptools@11.3:', when='@:2.1', type='build')
py-cx-oracle/package.py:    depends_on('py-setuptools@40.6.0:', type='build')
py-damask/package.py:    depends_on('py-setuptools@40.6:', type='build')
py-datalad/package.py:    depends_on('py-setuptools@40.8.0:', type='build')
py-dill/package.py:    depends_on('py-setuptools@0.6:', type='build')
py-dvc/package.py:    depends_on('py-setuptools@34.0.0:', type=('build', 'run'))
py-emcee/package.py:    depends_on('py-setuptools@40.6.0:', type='build', when='@3.1.1:')
py-extension-helpers/package.py:    depends_on('py-setuptools@30.3:', type='build')
py-fastcov/package.py:    depends_on('py-setuptools@38.3:', type='build')
py-fenics-ffcx/package.py:    depends_on('py-setuptools@58:', type='build', when="@0.4:")
py-flake8/package.py:    depends_on('py-setuptools@30:', type=('build', 'run'))
py-fs/package.py:    depends_on('py-setuptools@38.3.0:', type='build')
py-funcsigs/package.py:    depends_on('py-setuptools@17.1:', type='build')
py-geeup/package.py:    depends_on('py-setuptools@38.3.0:', type='build')
py-genshi/package.py:    depends_on("py-setuptools@:57", type='build')
py-gevent/package.py:    depends_on('py-setuptools@40.8:', when='@20.5.1:', type=('build', 'run'))
py-gevent/package.py:    depends_on('py-setuptools@40.8:', when='@1.5:', type='build')
py-gevent/package.py:    depends_on('py-setuptools@24.2:', when='@:1.4', type='build')
py-google-api-core/package.py:    depends_on('py-setuptools@34.0.0:', type=('build', 'run'))
py-google-auth/package.py:    depends_on('py-setuptools@40.3.0', type='build', when='@2.3.2:')
py-healpy/package.py:    depends_on('py-setuptools@3.2:', type='build')
py-hist/package.py:    depends_on('py-setuptools@45:', type='build')
py-histoprint/package.py:    depends_on('py-setuptools@42:',   type='build')
py-humanize/package.py:    depends_on('py-setuptools@42:', when='@3.2:', type=('build', 'run'))
py-hypothesis/package.py:    depends_on('py-setuptools@36.2:', type=('build'))
py-immutables/package.py:    depends_on('py-setuptools@42:', type='build', when='@0.16:')
py-importlib-metadata/package.py:    depends_on('py-setuptools@56:', type='build', when='@4.6.4:')
py-ipycanvas/package.py:    depends_on('py-setuptools@40.8:', type='build')
py-ipyevents/package.py:    depends_on('py-setuptools@40.8:', type='build')
py-ipyparallel/package.py:    depends_on('py-setuptools@40.8:', type='build', when='@7.1:')
py-ipython-cluster-helper/package.py:    depends_on('py-setuptools@18.5:', type=('build', 'run'))
py-ipython/package.py:    depends_on('py-setuptools@18.5:', type='run', when='@4.1:')
py-jsonpickle/package.py:    depends_on('py-setuptools@42:', type='build', when='@2.0.0:')
py-jupyter-console/package.py:    depends_on('py-setuptools@40.8.0:', type='build', when='@6.2:')
py-jupyter-packaging/package.py:    depends_on('py-setuptools@46.4:', when='@0.8:', type=('build', 'run'))
py-jupyter-packaging11/package.py:    depends_on('py-setuptools@46.4:', type=('build', 'run'))
py-jupyterlab-widgets/package.py:    depends_on('py-setuptools@40.8.0:', type='build')
py-keyring/package.py:    depends_on('py-setuptools@56:', when='@23.1:', type='build')
py-keyring/package.py:    depends_on('py-setuptools@42:', when='@21:', type='build')
py-keyring/package.py:    depends_on('py-setuptools@34.4:', type='build')
py-kubernetes/package.py:    depends_on('py-setuptools@21.0.0:', type=('build'))
py-lazy-object-proxy/package.py:    depends_on('py-setuptools@30.3.0:', type='build', when='@1.6.0:')
py-markdown/package.py:    depends_on('py-setuptools@36.6:', type='build', when='@3.1:')
py-meshio/package.py:    depends_on('py-setuptools@42:', type='build')
py-mg-rast-tools/package.py:    depends_on('py-setuptools@28.0:', type='build')
py-mock/package.py:    depends_on('py-setuptools@17.1:', type='build')
py-mplhep-data/package.py:    depends_on('py-setuptools@42:', type='build')
py-mplhep/package.py:    depends_on('py-setuptools@39.2:',      type='build')
py-mpmath/package.py:    depends_on('py-setuptools@36.7.0:', type='build', when='@1.2.0:')
py-multidict/package.py:    depends_on('py-setuptools@40:', type='build', when='@5.1.0:')
py-multiprocess/package.py:    depends_on('py-setuptools@0.6:', type='build')
py-nbdime/package.py:    depends_on('py-setuptools@40.8.0:',            type='build')
py-nest-asyncio/package.py:    depends_on('py-setuptools@42:', type='build')
py-neurokit2/package.py:    depends_on('py-setuptools@40.6.0:', type='build')
py-nibabel/package.py:    depends_on('py-setuptools@30.3.0:', type=('build', 'run'))
py-nibetaseries/package.py:    depends_on('py-setuptools@40.8:', type='build')
py-ninja/package.py:    depends_on('py-setuptools@42:', type='build')
py-nitransforms/package.py:    depends_on('py-setuptools@42.0:', type='build')
py-niworkflows/package.py:    depends_on('py-setuptools@38.4.1:', when='@1.2.3:', type='build')
py-niworkflows/package.py:    depends_on('py-setuptools@30.4.0:', type='build')
py-nose/package.py:    depends_on('py-setuptools@:57', type='build')
py-numcodecs/package.py:    depends_on('py-setuptools@18.1:', type='build')
py-numexpr3/package.py:    depends_on('py-setuptools@18.2:', type='build')
py-packaging/package.py:    depends_on('py-setuptools@40.8.0:', when='@20.8:', type='build')
py-pandas/package.py:    depends_on('py-setuptools@24.2.0:', type='build')
py-pandas/package.py:    depends_on('py-setuptools@38.6.0:', type='build', when='@1.3:')
py-pandas/package.py:    depends_on('py-setuptools@51.0.0:', type='build', when='@1.3.2:')
py-pathos/package.py:    depends_on('py-setuptools@0.6:', type='build')
py-platformdirs/package.py:    depends_on('py-setuptools@44:', type='build')
py-portalocker/package.py:    depends_on('py-setuptools@38.3.0:', type='build')
py-pox/package.py:    depends_on('py-setuptools@0.6:', type='build')
py-ppft/package.py:    depends_on('py-setuptools@0.6:', type='build')
py-pure-eval/package.py:    depends_on('py-setuptools@44:',  type='build')
py-pybids/package.py:    depends_on('py-setuptools@30.3.0:', type='build')
py-pycocotools/package.py:    depends_on('py-setuptools@18.0:', type='build')
py-pyfaidx/package.py:    depends_on('py-setuptools@0.7:', type=('build', 'run'))
py-pylev/package.py:    depends_on('py-setuptools@42:', type='build')
py-pylint/package.py:    depends_on('py-setuptools@17.1:', type='build')
py-pymatgen/package.py:    depends_on('py-setuptools@18.0:', type='build')
py-pyqt-builder/package.py:    depends_on('py-setuptools@30.3:', type='build')
py-pyrsistent/package.py:    depends_on('py-setuptools@42:', type='build', when='@0.18.0:')
py-pytest-check-links/package.py:    depends_on('py-setuptools@17.1:', type='build')
py-pytest-doctestplus/package.py:    depends_on('py-setuptools@30.3.0:', type=('build', 'run'))
py-pytest-openfiles/package.py:    depends_on('py-setuptools@30.3.1:', type='build')
py-pytest-runner/package.py:    depends_on('py-setuptools@42:', when='@5.3:', type=('build', 'run'))  # *
py-pytest-runner/package.py:    depends_on('py-setuptools@34.4:', when='@5:', type=('build', 'run'))  # *
py-pytest-runner/package.py:    depends_on('py-setuptools@27.3:', when='@4.1:', type=('build', 'run'))
py-pytest/package.py:    depends_on('py-setuptools@42.0:', when='@6.2:', type=('build', 'run'))
py-pytest/package.py:    depends_on('py-setuptools@40.0:', when='@3.9.2:6.1', type=('build', 'run'))
py-pytest/package.py:    depends_on('py-setuptools@30.3:', when='@3.9.0:3.9.1', type=('build', 'run'))
py-python-dateutil/package.py:    depends_on('py-setuptools@24.3:', type='build')
py-python-xlib/package.py:    depends_on('py-setuptools@30.3.0:', type='build')
py-qiskit-aer/package.py:    depends_on('py-setuptools@40.1.0:', type='build')
py-requests-futures/package.py:    depends_on('py-setuptools@38.6.1:', type='build')
py-ruamel-yaml-clib/package.py:    depends_on('py-setuptools@28.7.0:', type='build')
py-ruamel-yaml/package.py:    depends_on('py-setuptools@28.7:',        when='@0.17:',             type=('build'))
py-salib/package.py:    depends_on('py-setuptools@38.3:', type=('build', 'run'))
py-scikit-build/package.py:    depends_on('py-setuptools@28.0.0:', type=('build', 'run'))
py-scikit-image/package.py:    depends_on('py-setuptools@51:', when='@0.18:', type='build')
py-scipy/package.py:    depends_on('py-setuptools@:51.0.0', when='@1.6', type='build')
py-scipy/package.py:    depends_on('py-setuptools@:57', when='@1.7:', type='build')
py-secretstorage/package.py:    depends_on('py-setuptools@30.3:', type='build')
py-semantic-version/package.py:    depends_on('py-setuptools@0.8:', type='build')
py-setuptools-rust/package.py:    depends_on('py-setuptools@46.1:', when='@0.11.6:', type=('build', 'run'))
py-setuptools-scm/package.py:    depends_on('py-setuptools@34.4:', type=('build', 'run'))
py-setuptools-scm/package.py:    depends_on('py-setuptools@45:', type=('build', 'run'), when='@6:')
py-shtab/package.py:    depends_on('py-setuptools@42:', type='build')
py-singledispatch/package.py:    depends_on('py-setuptools@42:', type='build', when='@3.7:')
py-sip/package.py:    depends_on('py-setuptools@30.3:', when='@5:', type='build')
py-soupsieve/package.py:    depends_on('py-setuptools@42:', when='@2.2:', type='build')
py-spglib/package.py:    depends_on('py-setuptools@18.0:', type=('build', 'run'))
py-sphinx-autodoc-typehints/package.py:    depends_on('py-setuptools@40.0.4:', type='build')
py-statsmodels/package.py:    depends_on('py-setuptools@0.6c5:', type='build')
py-symfit/package.py:    depends_on('py-setuptools@17.1:', type='build')
py-templateflow/package.py:    depends_on('py-setuptools@40.9:', when='@0.7.1:', type='build')
py-templateflow/package.py:    depends_on('py-setuptools@30.3:', type='build')
py-tensorboard-plugin-wit/package.py:    depends_on('py-setuptools@36.2.0:', type='build')
py-tensorboard/package.py:    depends_on('py-setuptools@41.0.0:', type=('build', 'run'))
py-terminado/package.py:    depends_on('py-setuptools@40.8.0:', when='@0.12.1:', type='build')
py-torchgeo/package.py:    depends_on('py-setuptools@42:', type='build')
py-tqdm/package.py:    depends_on('py-setuptools@42:', type=('build', 'run'))
py-traitlets/package.py:    depends_on('py-setuptools@40.8:', type='build')
py-trojanzoo-sphinx-theme/package.py:    depends_on('py-setuptools@40.9:', type='build')
py-tweedledum/package.py:    depends_on('py-setuptools@42:', type='build')
py-twine/package.py:    depends_on('py-setuptools@0.7.0:', type=('build', 'run'))
py-twisted/package.py:    depends_on('py-setuptools@35.0.2:', type='build', when='@21.7.0:')
py-typeguard/package.py:    depends_on('py-setuptools@42:', type='build')
py-umi-tools/package.py:    depends_on('py-setuptools@1.1:',   type='build')
py-uproot/package.py:    depends_on('py-setuptools@42:', type='build', when='@4.1.8:')
py-vector/package.py:    depends_on('py-setuptools@42:',            type='build')
py-virtualenv/package.py:    depends_on('py-setuptools@40.6.3:', type=('build', 'run'))
py-virtualenv/package.py:    depends_on('py-setuptools@41.00.03:', type=('build', 'run'), when='@20.10.0:')
py-warpx/package.py:    depends_on('py-setuptools@42:', type='build')
py-wrapt/package.py:    depends_on('py-setuptools@38.3:', when='@1.13:', type='build')
py-xarray/package.py:    depends_on('py-setuptools@38.4:', when='@0.16:', type=('build', 'run'))
py-xarray/package.py:    depends_on('py-setuptools@40.4:', when='@0.17:', type=('build', 'run'))
py-yarl/package.py:    depends_on('py-setuptools@40:', type='build', when='@1.7.2:')
py-yt/package.py:    depends_on("py-setuptools@19.6:", type=('build', 'run'))
py-zarr/package.py:    depends_on('py-setuptools@18.0:',       type='build')
py-zarr/package.py:    depends_on('py-setuptools@38.6.0:',     type='build', when='@2.4.0:')
py-zc-buildout/package.py:    depends_on('py-setuptools@8.0:', type=('build', 'run'))
py-zipp/package.py:    depends_on('py-setuptools@34.4:', type='build', when='@0.3.3:')
py-zipp/package.py:    depends_on('py-setuptools@56:', type='build', when='@3.5.1:')
py-zope-interface/package.py:    depends_on('py-setuptools@:45',   type=('build', 'run'), when='@4.5.0')
```
</details>

Reply:
Then it is decided.